prepare_stmt.go

```diff
@@ -48,12 +46,18 @@ func (db *PreparedStmtDB) Close() {
 	db.Mux.Lock()
 	defer db.Mux.Unlock()
 
-	for _, query := range db.PreparedSQL {
-		if stmt, ok := db.Stmts[query]; ok {
-			delete(db.Stmts, query)
-			go stmt.Close()
-		}
+	for _, stmt := range db.Stmts {
+		go func(s *Stmt) {
+			// make sure the stmt must finish preparation first
+			<-s.prepared
+			if s.Stmt != nil {
+				_ = s.Close()
+			}
+		}(stmt)
 	}
+	// I think setting it to nil should be better, as there should be no one
+	// writing into it after calling Close, but leave it for compatibility and safety
+	db.Stmts = make(map[string]*Stmt)
```

Comment:
Setting db.Stmts to nil is ok fo me, because `PreparedStmtDB` should not be allowed to be used after `Close`